akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ServerSpec.scala

```diff
@@ -678,6 +685,32 @@ class Http2ServerSpec extends AkkaSpec("" + "akka.loglevel = debug")
       }
     }
 
+    "expose synthetic headers" should {
+      "expose Remote-Address" in new TestSetup with RequestResponseProbes with AutomaticHpackWireSupport {
+
+        lazy val theAddress = "127.0.0.1"
+        lazy val thePort = 1337
+        override def modifyServer(server: BidiFlow[HttpResponse, ByteString, ByteString, HttpRequest, NotUsed]) =
+          BidiFlow.fromGraph(StreamUtils.fuseAggressive(server).withAttributes(
+            HttpAttributes.remoteAddress(Some(new InetSocketAddress(theAddress, thePort)))
+          ))
+
+        val target = Uri("http://www.example.com/")
+        sendRequest(1, HttpRequest(uri = target))
+        requestIn.ensureSubscription()
+
+        val request = expectRequestRaw()
+        val remoteAddressHeader = request.header[headers.`Remote-Address`].get
+        remoteAddressHeader.address.getAddress.get().toString shouldBe ("/" + theAddress)
+        remoteAddressHeader.address.getPort shouldBe thePort
+
+        val streamIdHeader = request.header[Http2StreamIdHeader].getOrElse(Http2Compliance.missingHttpIdHeaderException)
```

Comment:
yup, ok; it does not really check things in relation to the synthetic header after all